MixItUp.Base/MixerAPI/InteractiveClientWrapper.cs

```diff
@@ -733,26 +731,34 @@ private async void Client_OnGiveInput(object sender, InteractiveGiveInputModel e
                         user = await ChannelSession.ActiveUsers.GetUserByID(e.participantID);
                         if (user == null)
                         {
-                            IEnumerable<InteractiveParticipantModel> recentParticipants = await this.GetRecentParticipants();
-                            InteractiveParticipantModel participant = recentParticipants.FirstOrDefault(p => p.sessionID.Equals(e.participantID));
-                            if (participant != null)
+                            InteractiveParticipantModel participant = null;
+                            if (this.Participants.TryGetValue(e.participantID, out participant))
+                            {
+                                user = new UserViewModel(participant);
+                            }
+                            else
                             {
-                                user = await ChannelSession.ActiveUsers.AddOrUpdateUser(participant);
+                                IEnumerable<InteractiveParticipantModel> recentParticipants = await this.GetRecentParticipants();
+                                participant = recentParticipants.FirstOrDefault(p => p.sessionID.Equals(e.participantID));
+                                if (participant != null)
+                                {
+                                    user = await ChannelSession.ActiveUsers.AddOrUpdateUser(participant);
+                                }
                             }
                         }
                     }
 
                     if (user == null)
                     {
-                        if (ChannelSession.Settings.PreventUnknownInteractiveUsers)
-                        {
-                            return;
-                        }
-
                         user = new UserViewModel(0, "Unknown User");
                         user.InteractiveIDs.Add(e.participantID);
                     }
 
+                    if (ChannelSession.Settings.PreventUnknownInteractiveUsers && user.ID == 0)
```

Comment:
Instead of user.ID == 0, consider adding a property? user.IsAnonymous? This would be clearer, and also allow us to "tweak" the anonymous logic later without having to search the code base for user.ID == 0.

Reply:
Then later, you can use it instead if: if (interactiveUser.userID > 0)

Reply:
The InteractiveParticipantModel is in the client library, so I don't want to put our logic in to there. That said, there is that new "anonymous" property that added which can also be used, but I feel the ID check is more important as that the anonymous property (though they'll both probably be 0 & true respectively at the same time, a user ID being 0 is a stronger indication of that user being unknown IMO)